hazelcast/src/main/java/com/hazelcast/instance/impl/DefaultNodeExtension.java

```diff
@@ -164,6 +165,12 @@ private void checkSecurityAllowed() {
                 throw new IllegalStateException("Symmetric Encryption requires Hazelcast Enterprise Edition");
             }
         }
+        AuditlogConfig auditlogConfig = node.getConfig().getAuditlogConfig();
+        if (auditlogConfig != null && auditlogConfig.isEnabled()) {
+            if (!BuildInfoProvider.getBuildInfo().isEnterprise()) {
+                throw new IllegalStateException("Auditlog requires Hazelcast Enterprise Edition");
```

Comment:
I guess we might also throw `InvalidConfigurationException` but then it would be different from the behaviour above. WDYT?

Reply:
This can be changed, but I would rather keep the reported exception in sync with other EE features (HR, persistence, security features).